File: sast-js/case/accuracy/context_sensitive/polymorphism/no_solver/polymorphism_001_T.js
// evaluation information start
// real case = true
// evaluation item = 准确度->上下文敏感分析->多态->无需通过对不同的条件进行求解，即能够区分不同的执行路径的状态
// scene introduction =
// level = 2
// bind_url = accuracy/context_sensitive/polymorphism/polymorphism_001_T
// evaluation information end
const { execSync } = require('child_process');

function polymorphism_001_T(__taint_src) {
  class Base { }

  class Sub1 extends Base {
    call() {
      return __taint_src;
    }
  }

  class Sub2 extends Base {
    call() {
      return "_";
    }
  }

  const sub = new Sub1();
  __taint_sink(sub.call());
}

function __taint_sink(o) {
  execSync(o);
}

const taint_src = "taint_src_value";

polymorphism_001_T(taint_src);

File: sast-go/cases/accuracy/context_sensitive/polymorphism/polymorphism_001_T/polymorphism_001_T.go

// evaluation information start
// real case = true
// evaluation item = 准确度->上下文敏感分析->多态
// scene introduction = 继承
// level = 2
// bind_url = accuracy/context_sensitive/polymorphism/polymorphism_001_T/polymorphism_001_T
// evaluation information end

package main

func polymorphism_001_T(__taint_src interface{}) {
	var sub Base
	sub = Sub1{}
	__taint_sink(sub.call(__taint_src))
}

type Base interface {
	call(src interface{}) interface{}
}

type Sub1 struct{}

func (s Sub1) call(src interface{}) interface{} {
	return src
}

type Sub2 struct{}

func (s Sub2) call(src interface{}) interface{} {
	return "_"
}

func __taint_sink(o interface{}) {
}
